src/main/flight/feedforward.c

```diff
@@ -64,103 +73,117 @@ void feedforwardInit(const pidProfile_t *pidProfile) {
+            // duplicateCount[axis] indicates number of prior duplicate/s
+
+            // calculate the jitter attenuator from average of two most recent abs rcCommand deltas vs jitter threshold
+            float jitterAttenuator = 1.0f;
+            if (feedforwardJitterFactor) {
+                if (rcCommandDelta < feedforwardJitterFactor) {
+                    jitterAttenuator = MAX(1.0f - ((rcCommandDelta + prevRcCommandDelta[axis]) / 2.0f) / feedforwardJitterFactor, 0.0f);
+                    jitterAttenuator = 1.0f - jitterAttenuator * jitterAttenuator;
+                }
             }
-            goodPacketCount[axis] = 0;
-        } else {
-            // we have movement
-            // count the number of valid steps to decide if we permit interpolation, for Tx ADC filter situations especially
-            if (goodPacketCount[axis] < 2) {
-                goodPacketCount[axis] += 1;
+
+            // reduce setpoint speed by half after a single duplicate or a third after two. Any more are forced to zero.
+            // needed because while sticks are moving, the next valid step up will be proportionally bigger
+            // and stops excessive feedforward where steps are at intervals, eg when the OpenTx ADC filter is active
+            // downside is that for truly held sticks, the first feedforward step won't be as big as it should be
+            if (duplicateCount[axis]) {
+                setpointSpeed /= duplicateCount[axis] + 1;
             }
-        }
 
-        prevSetpoint[axis] = setpoint;
+            // first order type smoothing for setpoint speed
+            setpointSpeed = prevSetpointSpeed[axis] + feedforwardSmoothFactor * (setpointSpeed - prevSetpointSpeed[axis]);
 
-        if (axis == FD_ROLL) {
-            DEBUG_SET(DEBUG_FEEDFORWARD, 0, lrintf(setpoint)); // setpoint after interpolations
-        }
+            // calculate acceleration from smoothed setpoint speed
+            setpointAcceleration = setpointSpeed - prevSetpointSpeed[axis];
+
+            // use rxRate to normalise acceleration to nominal RC packet interval of 100hz
+            // without this, we would get less boost than we should at higher Rx rates
+            // note rxRate shows the current packet interval, hence
+            // if no Rx packets are received for a period, boost amount is attenuated proportional to the delay 
+            setpointAcceleration *= rxRate * 0.01f;
+
+            // first order acceleration smoothing (with smoothed input, so second order all up)
+            setpointAcceleration = prevAcceleration[axis] + feedforwardSmoothFactor * (setpointAcceleration - prevAcceleration[axis]);
+
+            // jitter reduction to reduce acceleration steps at low rcCommand Delta values
+            // does not attenuate the basic feedforward amount, but this is small anyway at centre due to expo
+//            setpointAcceleration *= jitterAttenuator;
```

Comment:
Is this meant to be commented out? If so, removing is probably better.

Reply:
fixed, thanks